lib/module.js

```diff
@@ -164,24 +164,30 @@ export default function SentryModule (moduleOptions) {
 
   // Enable publishing of sourcemaps
   if (!options.disabled) {
-    this.extendBuild((config, { isClient, isModern, isDev }) => {
-      if (!options.publishRelease || isDev) {
+    this.nuxt.hook('webpack:config', (webpackConfigs) => {
+      if (!options.publishRelease || this.options.dev) {
```

Comment:
Using `this.options.dev` would change the logic. In old logic the WebpackPlugin was only added in one of the configs while with this it will be added to all configs. That would result in publishing release on building every configuration (server, client, modern).

Reply:
Hmm, actually that's not true. I guess it should be equivalent to old code as the configs should have devmode set to the same value as nuxt config.

Reply:
But then we don't even have to set up the hook if that first line of conditions is true.

Reply:
`isDev` is actually defined using `this.options.dev`

https://github.com/nuxt/nuxt.js/blob/e8aca9eb117851047e82e94948ff8b4bcb464b1a/packages/webpack/src/config/base.js#L38

Reply:
> But then we don't even have to set up the hook if that first line of conditions is true.

right

Reply:
fixed